### ISSUE TITLE
Jest exception handling

### ISSUE THREAD

**Issue by shavo007** [searchbox-io/Jest#361]
Hi guys,

My versions are:
 compile 'io.searchbox:jest:1.0.2'
    compile 'org.elasticsearch:elasticsearch:1.7.5'

I am testing a query and get a funny exception back:

java.lang.UnsupportedOperationException: JsonObject

I see that in io.searchbox.action.AbstractAction#createNewElasticSearchResult it does not even get as far as handing the response based in status code. It fails before.

This is not a meaningful exception.

For example, If i run this query using sense i get back the response:

```
{
   "error": {
      "root_cause": [
         {
            "type": "query_parsing_exception",
            "reason": "No value specified for term query",
            "index": "location-20160511142459",
            "line": 11,
            "col": 13
         }
      ],
      "type": "search_phase_execution_exception",
      "reason": "all shards failed",
      "phase": "query_fetch",
      "grouped": true,
      "failed_shards": [
         {
            "shard": 0,
            "index": "location-20160511142459",
            "node": "OJoaqd0-RBi5AzSOUmeqLg",
            "reason": {
               "type": "query_parsing_exception",
               "reason": "No value specified for term query",
               "index": "location-20160511142459",
               "line": 11,
               "col": 13
            }
         }
      ]
   },
   "status": 400
}


```

So based on any failed query, i will get back UnsupportedOperationException. But i prefer to log the more meaningful error above.

Is this fixed in later releases?

Thanks,
Shane.

**Comment by shavo007** [comment]
This is the line thats failing:

 errorMessage = jsonObject.get("error").getAsString();

**Comment by kramer** [comment]
This is fixed in jest versions 2.0 and later (see: #255) but was only recently backported to jest 1.x branch (see: #359) which is not released yet.

**Comment by shavo007** [comment]
Thanks for the reply. The above will give others info on what releases they can use.

Will close issue.
